### PR TITLE
proposal for lv_obj_swap() function

### PR DESCRIPTION
**Pull request by Karijn**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [ ] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [ ] Update the documentation

### CODE REVIEW

File: src/core/lv_obj_tree.c [hunk 1]
```diff
@@ -235,6 +235,27 @@ void lv_obj_move_background(lv_obj_t * obj)
...
+
+    lv_obj_t* parent = lv_obj_get_parent(obj1);
+    lv_obj_t* parent2 = lv_obj_get_parent(obj2);
+    LV_ASSERT(parent == parent2);
+
+    uint_fast32_t index1 = lv_obj_get_child_id(obj1);
+    uint_fast32_t index2 = lv_obj_get_child_id(obj2);
+
+    parent->spec_attr->children[index1] = obj2;
+    parent->spec_attr->children[index2] = obj1;
```
Comment (kisvegabor) [inline src/core/lv_obj_tree.c]: It should be `parent2->spec_attr->children[index2] = obj1;`, isn't it?

File: src/core/lv_obj_tree.c [hunk 2]
```diff
@@ -235,6 +235,27 @@ void lv_obj_move_background(lv_obj_t * obj)
...
+
+    lv_event_send(parent, LV_EVENT_CHILD_CHANGED, obj2);
```
Comment (kisvegabor) [inline src/core/lv_obj_tree.c:252]: `LV_EVENT_CHILD_CHANGED` should be sent to the parent.

It'd be better to send 2 events only if the parents are different.

File: src/core/lv_obj_tree.h [hunk 3]
```diff
@@ -101,6 +101,14 @@ void lv_obj_move_foreground(struct _lv_obj_t * obj);
  */
 void lv_obj_move_background(struct _lv_obj_t * obj);
 
+/**
+ * Swap two objects within the same parent.
```
Comment (kisvegabor) [inline src/core/lv_obj_tree.h]: It should work with different parents too.

**Review by kisvegabor — COMMENTED** [review]
Why have you closed the PR? It looked good in general, I had only a few minor comments.

File: examples/widgets/list/lv_example_list_2.c [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        uint_fast32_t i = lv_obj_get_child_id(currentButton);
+        if (i < lv_obj_get_child_cnt(parent) - 1)
+        {
+            lv_obj_swap(parent->spec_attr->children[i], parent->spec_attr->children[i + 1]);
+        }
+    }
+}
+
+
+void lv_example_list_2(void)
```
Comment (kisvegabor) [inline examples/widgets/list/lv_example_list_2.c:75]: It should be declared in [lv_example_widgets.h](https://github.com/lvgl/lvgl/blob/master/examples/widgets/lv_example_widgets.h).

File: examples/widgets/list/lv_example_list_2.c [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    lv_obj_set_pos(list2, 5, 5);
+    lv_obj_set_flex_flow(list2, LV_FLEX_FLOW_COLUMN);
+
+    /*Add buttons to the list*/
+    lv_obj_t* btn;
+
+    for (int i = 0; i < 10; i++)
+    {
+        char szBuf[100];
+        sprintf(szBuf, " Item %d ", i);
```
Comment (kisvegabor) [inline examples/widgets/list/lv_example_list_2.c]: Please use `lv_snprintf()` instead.

File: src/core/lv_obj_tree.c (hunk 1 above)
Comment (embeddedt) [reply src/core/lv_obj_tree.c]: It doesn't matter because `parent` and `parent2` are asserted to be equal a few lines up.

Comment (kisvegabor) [reply src/core/lv_obj_tree.c]: True. But why do they need to be equal? It seems to me this function can be used with different parents too.

File: examples/widgets/list/lv_example_list_2.c (hunk 5 above)
Comment (Karijn) [reply examples/widgets/list/lv_example_list_2.c]: ok

File: examples/widgets/list/lv_example_list_2.c (hunk 4 above)
Comment (Karijn) [reply examples/widgets/list/lv_example_list_2.c:75]: ok

**Comment by Karijn** [comment]
review issues applied

**Comment by embeddedt** [comment]
It seems that you keep closing it by accident; not sure why. You may be clicking "Close with comment" instead of "Comment"?

**Review by embeddedt — APPROVED** [review]

File: src/core/lv_obj_tree.h (hunk 3 above)
Comment (kisvegabor) [reply src/core/lv_obj_tree.h]: Thanks for the fixes. It seems only this comment should be updated.

Comment (Karijn) [reply src/core/lv_obj_tree.h]: ok

**Comment by kisvegabor** [comment]
Looks good, thank you very much!

I merge it.

**Pull request merged**